sympy/holonomic/holonomic.py

```diff
@@ -674,6 +702,155 @@ def __mul__(self, other):
 
         return HolonomicFunction(sol_ann, self.x)
 
+    __rmul__ = __mul__
+
+    def __sub__(self, other):
+        return self + other * -1
+
+    def __rsub__(self, other):
+        return self * -1 + other
+
+    def __pow__(self, n):
+
+        if n == 0:
+            return S(1)
+        if n == 1:
+            return self
+        else:
+            if n % 2 == 1:
+                powreduce = self**(n - 1)
+                return powreduce * self
+            elif n % 2 == 0:
+                powreduce = self**(n / 2)
+                return powreduce * powreduce
+
+    def composition(self, expr):
+        """
+        Returns the Holonomic Function after composition with
+        an algebraic function.
+
+        Examples
+        ========
+
+        >>> from sympy.holonomic.holonomic import HolonomicFunction, DiffOperatorAlgebra
+        >>> from sympy.polys.domains import ZZ, QQ
+        >>> from sympy import symbols
+        >>> x = symbols('x')
+        >>> R, Dx = DiffOperatorAlgebra(QQ.old_poly_ring(x),'Dx')
+
+        >>> HolonomicFunction(Dx - 1, x).composition(x**2) # e^(x**2)
+        HolonomicFunction((-2*x) + (1)Dx, x)
+
+        See Also
+        ========
+
+        From_Hyper
+        """
+
+        R = self.annihilator.parent
+        a = self.annihilator.order
+        diff = expr.diff()
+        listofpoly = self.annihilator.listofpoly
+
+        for i, j in enumerate(listofpoly):
+            if isinstance(j, self.annihilator.parent.base.dtype):
+                listofpoly[i] = self.annihilator.parent.base.to_sympy(j)
+
+        r = listofpoly[a].subs({self.x:expr})
+        subs = [-listofpoly[i].subs({self.x:expr}) / r for i in range (a)]
+        coeffs = [S(0) for i in range(a)] # Dkfa[i] == coeff of (D^i f)(a) in D^k (f(a))
```

Comment:
There should be at least 2 whitespaces before the beginning of an inline comment.